### ISSUE TITLE
shim: check syscall parameters for valid user address space

### ISSUE THREAD

**Issue by haraldh** [enarx/enarx#964]
If any syscall parameter has an address pointer, it must be checked for validity.

Fully implement the `sallyport` `validate*()` methods in all shims when `v2` of `sallyport` is used.

**Comment by platten** [comment]
Hey @rvolosatovs I’m using ZenHub in GitHub, [click this link](https://app.zenhub.com/workspaces/enarx-61d86a4144f52e001a7857d6/board) to join my workspace and see other features available in GitHub or [download the ZenHub extension](https://www.zenhub.com/extension) and sign up with your GitHub account. 
  _Posted using [ZenHub](https://www.zenhub.com)_

**Comment by rvolosatovs** [comment]
This happens by design in sallyport, what should happen in enarx for this feature to be closed? @haraldh 
Is this a duplicate of #1062 ?

**Comment by haraldh** [comment]
After #1062 this issue should be addressed. And no, it is not a duplicate.

**Comment by platten** [comment]
@haraldh @rvolosatovs Is this still relevant or should this be closed?

**Comment by haraldh** [comment]
yes
